x-pack/plugins/actions/server/builtin_action_types/lib/send_email.ts

```diff
@@ -50,12 +61,91 @@ export interface Content {
 // send an email
 export async function sendEmail(logger: Logger, options: SendEmailOptions): Promise<unknown> {
   const { transport, routing, content, configurationUtilities, hasAuth } = options;
-  const { service, host, port, secure, user, password } = transport;
+  const { service } = transport;
   const { from, to, cc, bcc } = routing;
   const { subject, message } = content;
 
+  const messageHTML = htmlFromMarkdown(logger, message);
+
+  if (service === AdditionalEmailServices.EXCHANGE) {
+    // request access token for microsoft exchange online server with Graph API scope
+    const res = await requestOAuthClientCredentialsToken(
+      `https://login.microsoftonline.com/${transport.tenantId}/oauth2/v2.0/token`,
```

Comment:
Do we need to handle a non 200 response here? The function returns `res.data` but is that guaranteed to have the fields we need?

Reply:
I think `logger.warn ` must be added, but not sure what else? Throw the excheption?

Reply:
I'm not sure. I'm guessing the task should fail (with appropriate message so it goes into event log) and a server log message written. I think throwing an exception will cause all of that to happen